### PR TITLE
Built a ReplyToDiscussions demo using Azure functions

### PR DESCRIPTION
**Pull request by seaki-msft**

### CODE REVIEW

File: samples/Functions-ReplyToDiscussions/HttpTrigger1.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace MyAmbassadorDemo.Function
+{
+    public static class HttpTrigger1
```
Comment (annaji-msft) [inline samples/Functions-ReplyToDiscussions/HttpTrigger1.cs:13]: Do we need this file when we already have timmer

File: samples/Functions-ReplyToDiscussions/TimerTrigger1.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+
+namespace MyAmbassadorDemo.Function
+{
+    public class TimerTrigger1
```
Comment (annaji-msft) [inline samples/Functions-ReplyToDiscussions/TimerTrigger1.cs:8]: name this more appropriately

File: samples/functions/replyToNewGithubDiscussion/ReplyToNewGithubDiscussion.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace APIMAuthorizationsDemo.Function
+{
+  public class ReplyToNewGithubDiscussion {
+    private const string APIM_SERVICE_NAME = "{APIM_SERVICE_NAME}";
```
Comment (annaji-msft) [inline samples/functions/replyToNewGithubDiscussion/ReplyToNewGithubDiscussion.cs:8]: can this be moved to app settings

File: samples/functions/replyToNewGithubDiscussion/ApimService.cs [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        private readonly string _subscriptionKey;
+        private readonly HttpClient _httpClient;
+        private readonly GraphQLHttpClient _graphQLClientForGithubEndpoints;
+
+        public ApimService(string baseUrl, string subscriptionKey, string identityToken = null)
+        {
+            _baseUrl = baseUrl;
+            _subscriptionKey = subscriptionKey;
+            _httpClient = new HttpClient();
+            _httpClient.DefaultRequestHeaders.Add("Ocp-Apim-Subscription-Key", subscriptionKey);
```
Comment (annaji-msft) [inline samples/functions/replyToNewGithubDiscussion/ApimService.cs:25]: move to app settings

**Review by annaji-msft — COMMENTED** [review]
In the read me would it help to add the steps to setup api management with the GitHub graphql and set the authorization policy?

File: samples/Functions-ReplyToDiscussions/HttpTrigger1.cs (hunk 1 above)
Comment (seandkim) [reply samples/Functions-ReplyToDiscussions/HttpTrigger1.cs]: Removed HttpTrigger

File: samples/functions/replyToNewGithubDiscussion/ReplyToNewGithubDiscussion.cs (hunk 3 above)
Comment (seandkim) [reply samples/functions/replyToNewGithubDiscussion/ReplyToNewGithubDiscussion.cs:8]: Moved to appsetting/environment var

File: samples/functions/replyToNewGithubDiscussion/ApimService.cs (hunk 4 above)
Comment (seandkim) [reply samples/functions/replyToNewGithubDiscussion/ApimService.cs:25]: moved

**Comment by seandkim** [comment]
> In the read me would it help to add the steps to setup api management with the GitHub graphql and set the authorization policy?

Wrote readme.md!

**Comment by seandkim** [comment]
I created a new PR https://github.com/Azure/APIManagement-Authorizations/pull/8 because I am not using this account anymore. Closing-

**Pull request closed without merging**